packages/sit-onyx/src/components/OnyxFlyoutMenu/future/OnyxFlyoutMenu.vue

```diff
@@ -24,23 +39,39 @@ const slots = defineSlots<{
   footer?(): unknown;
 }>();
 
+const open = useManagedState(
+  toRef(() => props.open),
+  false,
+  (val) => emit("update:open", val),
+);
+
 const {
-  elements: { button, menu, menuItem, listItem, flyout },
-  state: { isExpanded },
-} = createMenuButton({});
+  elements: { button, menu, menuItem, listItem },
+} = createMenuButton({ isExpanded: open, onToggle: () => (open.value = !open.value) });
+
+/**
+ * Debounced expanded state that will only be toggled after a given timeout.
+ */
+const updateDebouncedExpanded = debounce((expanded: boolean) => (open.value = expanded), 200);
 
 provide(MENU_BUTTON_ITEM_INJECTION_KEY, { menuItem, listItem });
 
 const { t } = injectI18n();
 </script>
 
 <template>
-  <div class="onyx-future-flyout-menu">
+  <!-- eslint-disable-next-line vuejs-accessibility/no-static-element-interactions -->
+  <div
+    class="onyx-future-flyout-menu"
+    @focusin="open = true"
+    @focusout="open = false"
+    @mouseover="() => updateDebouncedExpanded(true)"
+    @mouseout="() => updateDebouncedExpanded(false)"
```

Comment:
Why did you move this out of the composable?

Reply:
see PR description: These are features that are not in the pattern or spec.